rasa/nlu/extractors/entity_synonyms.py

```diff
@@ -165,3 +165,10 @@ def _add_entities_if_synonyms(
                     )
 
                 self.synonyms[original] = replacement
+
+            # add a self-reference to handle entities extracted in alternate cases
+            # i.e. for a synonym Austria,
+            # entities extracted as AUSTRIA, austria, ausTRIA, etc
+            # should also have the value of `Austria`
+            if replacement not in self.synonyms:
```

Comment:
Shouldn't this be `if original.lower() == replacement.lower() and replacement not in self.synonyms.values()` in order to add the self reference to the synonym title? e.g. the example in the ticket where `cashier check` should be mapped too:
```
synonym: cashier check
examples: |
  - cashier check
  - cashier's check
  - Cashier checks
```
I think this check also makes it explicit and more readable.

Reply:
> replacement not in self.synonyms.values() will be mostly false ?

Why would it be false? 🤔

Reply:
I have to admit I'm super confused. We might need some refactoring to improve readability, like better variable names than `original` and `replacement` (maybe with `synonym_key`, respectively `synonym_value` instead?). To double-check, in the example above with the synonym title `cashier check`, is this title the `original` and the examples can each be a `replacement`? 
I think you are right about the equivalence, but let's name variables to reduce confusion 😅 `replacement.lower()` should be `case_insensitive_synonym_key`?

Reply:
> To double-check, in the example above with the synonym title cashier check, is this title the original and the examples can each be a replacement?

It is the other way. Synonym title is the `replacement` that replaces the `original` entity from synonyms examples. For example, this synonym definition

```
- synonym: austria
  examples: |
    - vienna
    - salzburg
    - innsbruck
```

results in the a dictionary of this kind,

```
{
  "vienna": "austria",
  "salzburg": "austria",
  "innsbruck": "austria",
  "austria": "austria",
}
```

Do you think adding a comment to clarify `replacement` and `original` will make it more readable or should I change the variable names?

Reply:
Oh right, now this is clearer. Yes, let's add a comment and/or docstring as well 👍🏻 I'd still assign `replacement.lower()` to a variable since you're using it twice.

Reply:
Perfect 👍🏻